src/libutil/posix-source-accessor.cc

```diff
@@ -114,7 +114,10 @@ std::optional<SourceAccessor::Stat> PosixSourceAccessor::maybeLstat(const CanonP
     auto st = cachedLstat(path);
     if (!st)
         return std::nullopt;
-    mtime = std::max(mtime, st->st_mtime);
```

Comment:
We use mtime in `src/libfetchers/path.cc`, not sure if this is a data race, however we use posix source accessor in threads.